Package.swift

```diff
@@ -2,43 +2,68 @@
 
 import PackageDescription
 
+#if compiler(>=6.0)
+// this can be further improved once SE-0435 is available
+let swift5Mode = SwiftSetting.unsafeFlags(["-swift-version", "5"])
+let swift6Mode = SwiftSetting.unsafeFlags(["-swift-version", "6"])
+let toolsSwiftMode = swift6Mode
+let baseSettings: [SwiftSetting] = [
+]
+#else
+let swift5Mode = SwiftSetting.unsafeFlags(["-swift-version", "5"])
+let swift6Mode = SwiftSetting.unsafeFlags(["-swift-version", "6"])
```

Comment:
Do these variables need to be included in each condition, or can they be lifted above? They appear to be the same.

Reply:
Today they *must* be the same because there doesn't yet appear to be a Swift 6 snapshot that allows me to use `SwiftSetting. swiftLanguageVersion(.v6)`. I thought it could be nice to express the setting more semantically in that case.

I just checked the June 4 build and it still isn't working. But I also cannot tell *why* isn't working because of https://github.com/apple/swift/issues/74081

Reply:
I'm just wondering why it isn't written like this

```swift
let swift5Mode = SwiftSetting.unsafeFlags(["-swift-version", "5"])
let swift6Mode = SwiftSetting.unsafeFlags(["-swift-version", "6"])

#if compiler(>=6.0)
// this can be further improved once SE-0435 is available
let toolsSwiftMode = swift6Mode
let baseSettings: [SwiftSetting] = [
]
#else
let toolsSwiftMode = swift5Mode
let baseSettings: [SwiftSetting] = [
	.enableExperimentalFeature("StrictConcurrency")
]
#endif
```

Reply:
I was kinda hoping that eventually it could look like this to better highlight SE-0435. But, maybe this is silly. It does make it less concise...

```swift
#if compiler(>=6.0)
let swift5Mode = SwiftSetting.swiftLanguageVersion(.v5)
let swift6Mode = SwiftSetting.swiftLanguageVersion(.v5)
let toolsSwiftMode = swift6Mode
let baseSettings: [SwiftSetting] = [
]
#else
let swift5Mode = SwiftSetting.unsafeFlags(["-swift-version", "5"])
let swift6Mode = SwiftSetting.unsafeFlags(["-swift-version", "6"])
let toolsSwiftMode = swift5Mode
let baseSettings: [SwiftSetting] = [
	.enableExperimentalFeature("StrictConcurrency")
]
#endif
```

Reply:
Ahh I understand!

Reply:
No I'm wrong about this. We cannot ever simultaneously use a 6.0 tools feature and also declare compatibility with 5.10.

Reply:
I thought more on this. I think having Swift 5 targets is very useful, but I just don't see what we get by maintaining compatibility with Swift 5.10 tools. Plus it makes the package so much simpler.